CMakeLists.txt

```diff
@@ -80,11 +76,19 @@ target_include_directories(trieste
 
 target_link_libraries(trieste
   INTERFACE
-  snmallocshim-static
   re2::re2
   CLI11::CLI11
+  snmalloc
```

Comment:
Just for my own understanding, why does Trieste need to link against snmalloc even if it doesn't use it?

Reply:
So Trieste uses some of snmallocs headers.  There are a bunch of cross platform defines that are useful.  `SNMALLOC_FAST_PATH` and `SNMALLOC_SLOW_PATH` which I use in `snmalloc` heavily to get the code gen to inline precisely how I need it to.  I use that in the core of the rewrite engine to similarly get the code gen I was looking for.

That is not linking against any binaries with that target, only the header files.

Reply:
Sorry, it was actually in the Logging library, but for code gen purposes.